src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -129,7 +145,24 @@ let pp_arg ppf (custom_scalar_opt, (_, name, ut)) =
     | Some scalar -> scalar
     | None -> stantype_prim_str ut
   in
-  pf ppf "const %a& %s" pp_unsizedtype_custom_scalar (scalar, ut) name
+  (* we add the _arg suffix for any Eigen types *)
+  pf ppf "const %a& %s" pp_unsizedtype_custom_scalar_eigen_exprs (scalar, ut)
+    name
+
+let pp_arg_eigen_suffix ppf (custom_scalar_opt, (_, name, ut)) =
+  let scalar =
+    match custom_scalar_opt with
+    | Some scalar -> scalar
+    | None -> stantype_prim_str ut
+  in
+  (* we add the _arg suffix for any Eigen types *)
+  let opt_arg_suffix =
+    match ut with
+    | UMatrix | URowVector | UVector -> name ^ "_arg__"
+    | _ -> name
```

Comment:
It's more consistent with rest of the PR to use `if UnsizedType.is_eigen_type ut` rather than a `match`.